src/_Common/Functions.cs

```diff
@@ -105,6 +105,36 @@ internal static TimeSpan ToTimeSpan(this PeriodSize periodSize)
             };
         }
 
+        internal static PeriodSize? ToPeriodSize(this TimeSpan timeSpan)
+        {
+            PeriodSize? _periodSize = null;
+
+            if (timeSpan == TimeSpan.FromMinutes(1))
```

Comment:
If this doesn't support switch, it might be easier to just delete `_periodSize` and just do:

```csharp
if (timeSpan == TimeSpan.FromMinutes(1))
  return PeriodSize.OneMinute;
```
This will avoid subsequent evaluations.

Reply:
I started developing as you suggested, however there is always a need for a default return otherwise it won't compile. Otherwise I could use a null retrun, which only occurs if the precedents are not satisfied, can it go?

`if (timeSpan == TimeSpan.FromMinutes(1))
  return PeriodSize.OneMinute;
....

return null`

Reply:
Yes, adding a `return null;` at the end would satisfy that need.